src/components/ui/EditableFormCard.tsx

```diff
@@ -102,6 +107,18 @@ export default function EditableFormCard({ form, readonly = false }: UpdateFormP
   const onSubmit = async (data: FormDtoSchema) => {
     if (readonly) return;
 
+    if (allFormsData) {
+      const duplicateTitle = allFormsData.find((form: IFormType) => form.name === data.name);
+      if (duplicateTitle) {
+        handleShowAlert(dispatch, {
+          type: AlertType.Error,
+          message: "A form with the same title already exists",
+        });
+        return;
+      }
+     
+    }
+
```

Comment:
- I think we didn't need to check to see if we have `allFormsData` because if we don't have it `duplicateTitle` will be `undefined` and the `if(duplicate)` will handle that anyways.
- I also think we don't need that `return` after calling `handleShowAlert`

 
```suggestion
    const duplicateTitle = allFormsData.find(
      (form: IFormType) => form.name === data.name,
    )

    if (duplicateTitle) {
      handleShowAlert(dispatch, {
        type: AlertType.Error,
        message: "A form with the same title already exists",
      })
    }
```